Framework/Kernel/src/FacilityInfo.cpp

```diff
@@ -187,6 +187,19 @@ const InstrumentInfo &FacilityInfo::instrument(std::string iName) const {
                               [&iName](const auto &inst) { return boost::iequals(inst.shortName(), iName); });
   }
 
+  // Cover edge case where delimiter is present
+  if (instrument == m_instruments.cend()) {
+    instrument = std::find_if(m_instruments.cbegin(), m_instruments.cend(), [&iName](const auto &inst) {
+      return boost::iequals(inst.name() + inst.delimiter(), iName);
+    });
+  }
+
+  if (instrument == m_instruments.cend()) {
+    instrument = std::find_if(m_instruments.cbegin(), m_instruments.cend(), [&iName](const auto &inst) {
+      return boost::iequals(inst.shortName() + inst.delimiter(), iName);
+    });
+  }
+
```

Comment:
I have a strong suspicion this is not needed, because I remember that FileFinder already has its own getInstrument function that includes this check for the delimiter. So I can remove this change and use the FileFinder's function instead, which also makes more sense.

Reply:
Took it out without changing functunality